src/module.c

```diff
@@ -11244,6 +11442,30 @@ dictType sdsKeyValueHashDictType = {
     NULL                        /* allow to expand */
 };
 
+uint64_t dictModuleObjHash(const void *key) {
+    const RedisModule *o = key;
+    return dictGenHashFunction(o->name, sdslen((sds)o->name));
+}
+
+int dictModuleObjectCompare(dict *d, const void *key1, const void *key2) {
+    UNUSED(d);
+    const RedisModule *o1 = (RedisModule*) key1, *o2 = (RedisModule*) key2;
+    return sdscmp((sds) o1->name, (sds) o2->name);
```

Comment:
`return sdscmp((sds) o1->name, (sds) o2->name) == 0;` ?

Reply:
From your above comment, IIUC this method won't be required further. Let me test it out.